surface/include/pcl/surface/impl/marching_cubes.hpp

```diff
@@ -302,17 +273,24 @@ pcl::MarchingCubes<PointNT>::performReconstruction (pcl::PointCloud<PointNT> &po
 
   // Run the actual marching cubes algorithm, store it into a point cloud,
   // and copy the point cloud + connectivity into output
-  points.clear ();
+  intermediate_cloud.clear();
+
+  // preallocate memory assuming a hull. suppose 6 point per voxel
+  intermediate_cloud.reserve((size_t)(res_y_*res_z_ + res_x_*res_z_ + res_x_*res_y_) * 2 * 6);
```

Comment:
I meant that this product may exceed what `size_t` can represent. In this case we can actually get some small number as a result and reserve too less. I'd propose to perform this multiplication in double, then clamp to [`intermediate_cloud.max_size()`](http://en.cppreference.com/w/cpp/container/vector/max_size), and finally reserve.